[WIP] Desktop: Add Unit Tests

rihardsgravis opened the pull request:
# Description

Add Unit Tests for:

- [x] React UI components
- [x] React UI views visual
- [x] Native library wrappers
- [x] SeedStore classes
- [x] End-to-end tests for:
  - [x] Onboarding existing seed 
  - [x] Onboarding new seed
  - [ ] Send & receive

## Type of change

- Enhancement (a non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

Commit:
Initial testing coverage

Commit:
Add UI component tests

Commit:
Revert `file-loader` dependency removal

Commit:
Merge branch 'develop' into desktop-test-coverage

* develop:
  Update garbage collector usage (#885)
  Bump electron-updater from 4.0.4 to 4.0.6 in /src/desktop (#943)
  Increase timeout for attachToTangle network requests (#989)

# Conflicts:
#	src/desktop/npm-shrinkwrap.json

rajivshah3 commented on src/desktop/package.json:
```diff
@@ -51,13 +54,18 @@
     "express": "^4.16.3",
     "file-loader": "^2.0.0",
     "html-webpack-plugin": "^3.2.0",
+    "jest": "^23.6.0",
```
`jest` and `babel-jest` are bringing in vulnerable versions of `braces`, which is causing the Snyk test to fail. We need to force `braces@^2.3.1` to solve this issue. A package like https://www.npmjs.com/package/npm-force-resolutions might help, but that seems to work for lockfiles and not shrinkwraps. If I can figure out how to make it work for shrinkwraps I'll make a PR

rihardsgravis replied:
You can achieve that by editing the shrinkwrap file manually and change the child dependency version there.

rajivshah3 replied:
Would it be overwritten by `npm install`?

rihardsgravis replied:
It should not be overwritten - it's the opposite - `npm install` uses the shrinkwrap file to resolve the exact package versions (the only reason to have a shrinkwrap file at all).

rihardsgravis replied:
So, jest released `24.0.0` two days ago. We are ok now 👍

Commit:
Linting fixes

rihardsgravis committed:
Merge branch 'develop' into desktop-test-coverage

Commit:
Jest dependency update

Commit:
Lint fixes for shared codebase

Commit:
- Remove mobile jest `roots` config
- Add desktop tests to `tests:all`

Commit:
Add input component unit tests

Commit:
Update UI component tests

Commit:
Add crypto lib tests

Commit:
Add seedVault tests

Commit:
Merge branch 'develop' into desktop-test-coverage

* develop:
  Bump keytar from 4.3.0 to 4.3.1 in /src/desktop (#1003)
  Mobile/Android: Remove unused native crypto module (#1001)
  Bump fastlane from 2.114.0 to 2.115.0 in /src/mobile/android (#996)
  Bump fastlane from 2.114.0 to 2.115.0 in /src/mobile/ios (#997)

# Conflicts:
#	src/desktop/npm-shrinkwrap.json

Commit:
Add initial preload script tests

Commit:
Update screenshot mock creator

Commit:
Initial visual UI testing implementation

Commit:
Merge branch 'develop' into desktop-test-coverage

* develop:
  Bump @babel/plugin-proposal-object-rest-spread in /src/desktop (#1015)
  Bump webpack from 4.29.0 to 4.29.1 in /src/desktop (#1014)
  Bump dart-sass from 1.16.1 to 1.17.0 in /src/desktop (#1016)
  Bump webpack-cli from 3.2.1 to 3.2.3 in /src/desktop (#1017)
  Bump puppeteer from 1.11.0 to 1.12.1 in /src/desktop (#1010)
  Mobile: [Snyk] Fix for 2 vulnerable dependencies (#1012)
  Mobile: Bump build numbers for release v0.6.1 (40)
  Bump keytar from 4.3.1 to 4.3.2 in /src/desktop (#1007)
  Bump electron from 3.1.2 to 3.1.3 in /src/desktop (#1008)
  [Security] Bump sshpk from 1.10.2 to 1.16.1 in /src/desktop (#1006)
  Bump react-i18next from 9.0.9 to 9.0.10 in /src/desktop (#1002)

# Conflicts:
#	src/desktop/npm-shrinkwrap.json
#	src/mobile/package.json

Commit:
Add missing view visual snapshot tests

rajivshah3 commented:
@rihardsgravis I resolved the vulnerable dependency issue in https://github.com/facebook-atom/jest-electron-runner/commit/6f3e2979d7076171d7263517b34889036e384953. Just waiting for them to create a new release so the Snyk tests will pass

rajivshah3 committed:
Bump @jest-runner/electron to 1.1.0

Commit:
Merge branch 'develop' into desktop-test-coverage

* develop: (155 commits)
  Bump electron from 3.1.6 to 3.1.8 in /src/desktop (#1294)
  Bump react-transition-group from 2.7.1 to 2.9.0 in /src/desktop (#1273)
  Bump css-loader from 2.1.0 to 2.1.1 in /src/desktop (#1262)
  Bump electron-builder from 20.38.5 to 20.39.0 in /src/desktop (#1261)
  Bump @ledgerhq/hw-transport-node-hid in /src/desktop (#1290)
  Bump snyk from 1.149.0 to 1.150.0 (#1289)
  Bump fastlane from 2.119.0 to 2.120.0 in /src/mobile/ios (#1288)
  Bump fastlane from 2.119.0 to 2.120.0 in /src/mobile/android (#1287)
  Bump snyk from 1.148.0 to 1.149.0 (#1286)
  Shared: Add node request timeouts for getTransactionsToApprove endpoint (#1284)
  Mobile: Wait for SeedStore constructor promise to resolve before making a retry attempt (#1283)
  Bump snyk from 1.147.4 to 1.148.0 (#1279)
  Mobile: Minor changes to seed text input (#1244)
  New Crowdin translations (#1252)
  Mobile: Reenable and refactor deep linking (#1143)
  Only convert transactions to normalised transactions if notification function is defined (#1276)
  feat: add realm migration from v1 to v2 (#1274)
  Mobile: Fix AsyncStorage migration (#1272)
  Bump webpack-dev-middleware from 3.6.0 to 3.6.2 in /src/desktop (#1264)
  Bump snyk from 1.147.3 to 1.147.4 (#1270)
  ...

# Conflicts:
#	package.json
#	src/desktop/native/Entangled.js
#	src/desktop/package.json
#	src/desktop/src/index.js
#	src/desktop/src/libs/crypto.js
#	src/desktop/src/ui/components/Balance.js
#	src/desktop/src/ui/components/List.js
#	src/desktop/src/ui/global/Theme.js
#	src/desktop/src/ui/views/settings/account/Addresses.js
#	src/desktop/src/ui/views/wallet/Sidebar.js
#	src/shared/libs/currency.js

Commit:
Update Electron mock for tests

Commit:
Merge branch 'develop' into desktop-test-coverage

* develop: (125 commits)
  Bump snyk from 1.161.0 to 1.161.1 (#1455)
  Bump nyc from 13.1.0 to 14.1.0 in /src/shared (#1451)
  Bump snyk from 1.158.0 to 1.161.0 (#1448)
  Bump husky from 2.1.0 to 2.2.0 (#1447)
  Fix Electron API usage for Windows deep link functionality (#1438)
  Mobile: Bump build no to 61 (#1443)
  Mobile: Fix crash on empty password fields (#1441)
  Mobile: Bump build to 60 (#1437)
  New Crowdin translations (#1428)
  Convert all nested properties of address data from RealmObject to plain object (#1431)
  Shared: Add SeedVault export validation check (#1432)
  Mobile: Fix Android splash screen hang when returning from bg (#1429)
  Shared: Fix milestone fall behind check (#1436)
  Bump @babel/plugin-proposal-class-properties in /src/desktop (#1425)
  Bump @babel/preset-env from 7.4.3 to 7.4.4 in /src/desktop (#1415)
  [Security] Bump tar from 2.2.1 to 4.4.8 in /src/desktop (#1434)
  Bump snyk from 1.156.0 to 1.158.0 (#1435)
  Bump snyk from 1.154.0 to 1.156.0 (#1433)
  Bump @babel/register from 7.4.0 to 7.4.4 in /src/desktop (#1426)
  Bump snyk from 1.153.0 to 1.154.0 (#1424)
  ...

# Conflicts:
#	src/desktop/npm-shrinkwrap.json
#	src/desktop/package.json
#	src/desktop/src/libs/crypto.js
#	src/desktop/src/ui/components/List.js
#	src/desktop/src/ui/global/SeedExport.js
#	src/shared/libs/iota/utils.js

Commit:
Merge branch 'develop' into desktop-test-coverage

* develop: (23 commits)
  Bump snyk from 1.162.0 to 1.163.0 (#1491)
  New Crowdin translations (#1446)
  Desktop: Add migration explanation (#1445)
  Add issue template metadata [ci skip] (#1486)
  Update issue templates to newer version (#1484)
  Shared: Bump update versioning (#1478)
  Bump snyk from 1.161.1 to 1.162.0 in /src/shared (#1482)
  Bump snyk from 1.161.1 to 1.162.0 (#1481)
  Bump @babel/plugin-proposal-export-namespace-from in /src/shared (#1450)
  Desktop 0.5.0 Release (#1479)
  Desktop: Settings and Onboarding fixes (#1457)
  Mobile/Android: Rebuild Entangled libs (#1464)
  Bump dart-sass from 1.19.0 to 1.20.1 in /src/desktop (#1469)
  Bump webpack-cli from 3.3.1 to 3.3.2 in /src/desktop (#1472)
  Bump electron from 3.1.8 to 3.1.9 in /src/desktop (#1475)
  Shared: [Snyk] Fix for 1 vulnerable dependencies (#1474)
  Mobile: [Snyk] Fix for 1 vulnerable dependencies (#1473)
  Bump eslint-plugin-react-native from 3.2.1 to 3.7.0 (#1389)
  Bump fastlane from 2.121.1 to 2.122.0 in /src/mobile/android (#1468)
  Bump fastlane from 2.121.1 to 2.122.0 in /src/mobile/ios (#1467)
  ...

Commit:
- Update Electron and state mocks
- Skip using use webpack dev server for tests
- Update Bugsnag usage
- Update Lodash imports usage

Commit:
Remove map file upload to Bugsnag

Commit:
Disable Tray devtools by default

Commit:
Add Trinity onboarding (existing seed) end-to-end test

Commit:
Merge branch 'develop' into desktop-test-coverage

* develop:
  Bump react-dev-utils from 9.0.0 to 9.0.1 in /src/desktop (#1490)
  Bump electron-builder from 20.39.0 to 20.40.2 in /src/desktop (#1496)
  Bump webpack from 4.30.0 to 4.31.0 in /src/desktop (#1500)
  Shared: Fix extra parameter in fn call when reattaching and promoting (#1503)
  Mobile: Bump handlebars to 4.1.2 (#1454)
  Mobile/iOS: Patch react-native run-ios (#1505)
  Bump url-parse from 1.4.3 to 1.4.7 in /src/shared (#1453)
  Bump proxy-polyfill from 9c009d7 to 0.3.0 in /src/shared (#1449)
  Bump snyk from 1.163.1 to 1.163.3 (#1499)
  Bump react-transition-group from 4.0.0 to 4.0.1 in /src/desktop (#1497)
  - Update Bugsnag dependency usage (#1493)
  Bump snyk from 1.163.0 to 1.163.1 (#1495)

# Conflicts:
#	src/desktop/bugsnag.js
#	src/desktop/npm-shrinkwrap.json
#	src/desktop/package.json
#	src/shared/store.js

Commit:
Update shrinkwrap

Commit:
Add Onboarding new seed e2e test

Commit:
Clean up test mocks
